### PR TITLE
cmd/acme: Add support for ECDSA private keys

### PR DESCRIPTION
**Pull request by titanous**

### CODE REVIEW

File: cmd/acme/config.go [hunk 1]
```diff
@@ -37,6 +38,8 @@ const (
 
 	// rsaPrivateKey is a type of RSA key.
 	rsaPrivateKey = "RSA PRIVATE KEY"
+
+	ecPrivateKey = "EC PRIVATE KEY"
```
Comment (crhym3) [inline cmd/acme/config.go]: I'd remove the empty line below and // rsaPrivateKey comment altogether then.

File: cmd/acme/config.go [hunk 2]
```diff
@@ -134,19 +141,19 @@ func writeKey(path string, k *rsa.PrivateKey) error {
...
+func anyKey(filename string, gen bool) (crypto.Signer, error) {
 	k, err := readKey(filename)
 	if err == nil {
 		return k, nil
 	}
 	if !os.IsNotExist(err) || !gen {
 		return nil, err
 	}
-	k, err = rsa.GenerateKey(rand.Reader, 2048)
+	rsaKey, err := rsa.GenerateKey(rand.Reader, 2048)
```
Comment (crhym3) [inline cmd/acme/config.go]: maybe default to the EC then? seems like they are quite popular these days.

**Comment by crhym3** [comment]
Reviewed. Been meaning to do this for quite some time. Thanks!

**Comment by titanous** [comment]
Comments addressed.

**Comment by crhym3** [comment]
Some help text also needs to change: https://github.com/titanous/acme/blob/98c87fa04d38f4cef0cafc30ddc6b7e2bb701681/cmd/acme/usage.go#L40

Also, https://github.com/titanous/acme/blob/98c87fa04d38f4cef0cafc30ddc6b7e2bb701681/cmd/acme/reg.go#L30 - there's no mention what `-gen` flag actually does, maybe it's time to add it there, too?

I'd say which key types are supported and what key type the program will generate by default.

**Comment by titanous** [comment]
I've updated the usage.

**Comment by crhym3** [comment]
Great. Thanks!

**Pull request merged**